### PR TITLE
Add recipe for google-cloud-cpp and dependencies

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by xhochy** [comment]
@conda-forge/staged-recipes This is ready for review. Once merged, I will also make a PR to https://github.com/conda-forge/conda-forge-pinning-feedstock to have global pins we can migrate.

File: recipes/crc32-cpp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{% set name = "crc32c" %}
+{% set version = "1.1.1" %}
+
+package:
+  name: {{ name|lower }}-cpp
```
Comment (isuruf) [inline recipes/crc32-cpp/meta.yaml]: Why a `-cpp` name?

Comment (xhochy) [reply recipes/crc32-cpp/meta.yaml]: We have a `crc32c` Python package on conda-forge already and this is the C++ implementation of `crc32c`.

Comment (isuruf) [reply recipes/crc32-cpp/meta.yaml]: Ah. okay. How about `libcrc32c`?

Comment (xhochy) [reply recipes/crc32-cpp/meta.yaml]: Done and CI passed.

**Comment by wesm** [comment]
Sweet thanks @xhochy!

**Pull request merged**